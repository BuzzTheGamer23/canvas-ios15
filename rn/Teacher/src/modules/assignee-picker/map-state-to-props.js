// @flow

export type AssigneeSearchProps = {
  courseID: string,
  sections: Section[],
  enrollments: Enrollment[],
  onSelection: Function,
  navigator: ReactNavigator,
  refreshSections: Function,
  refreshEnrollments: Function,
}

export default function mapStateToProps (state: AppState, ownProps: AssigneeSearchProps): any {
  const courseID = ownProps.courseID
  const enrollments = Object.values(state.entities.enrollments).filter((item) => {
    // $FlowFixMe
    if (item.course_id !== courseID) return false
    // $FlowFixMe
    if (item.type !== 'StudentEnrollment') return false
    return true
  })

  const sections = Object.values(state.entities.sections).filter((item) => {
    // $FlowFixMe
    return item.course_id === courseID
  })

  return {
    sections,
    enrollments,
  }
}
