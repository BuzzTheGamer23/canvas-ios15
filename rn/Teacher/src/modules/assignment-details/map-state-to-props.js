// @flow

export type AssignmentDetailsState = {
  +assignmentDetails: Assignment,
  +pending: number,
  +error?: string,
}

export type AssignmentDetailsProps = {
  assignmentDetails: Assignment,
  navigator: ReactNavigator,
  courseID: string,
  assignmentID: string,
  error?: string,
  pending?: number,
  refresh: Function,
}

export type AssignmentDetailsActionProps = {
  +refreshAssignmentDetails: (courseID: number, assignmentID: string) => Promise<Assignment>,
}

export function mapStateToProps (state: AppState, ownProps: AssignmentDetailsProps): AssignmentDetailsState {
  const assignment = state.entities.assignments[ownProps.assignmentID]

  return {
    assignmentDetails: assignment,
    ...assignment,
    pending: state.entities.courses[ownProps.courseID].assignmentGroups.pending,
  }
}

