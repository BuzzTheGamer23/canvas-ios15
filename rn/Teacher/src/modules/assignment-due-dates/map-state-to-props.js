// @flow

import AssignmentDates from '../../common/AssignmentDates'
import Navigator from '../../routing/Navigator'

export type AssignmentDueDatesState = {
  +assignment: Assignment,
  +users: {},
  +courseColor: string,
}

export type AssignmentDueDatesProps = {
  courseID: string,
  assignmentID: string,
  assignment: Assignment,
  users: {},
  refreshUsers: Function,
  navigator: Navigator,
  courseColor: string,
  quizID?: ?string,
}

export type AssignmentDueDatesActionProps = {
  +refreshUsers: () => Promise<User[]>,
}

export function mapStateToProps (state: AppState, ownProps: AssignmentDueDatesProps): AssignmentDueDatesState {
  const assignment = state.entities.assignments[ownProps.assignmentID].data
  const dates = new AssignmentDates(assignment)
  const users = {}
  const courseColor = state.entities.courses[ownProps.courseID].color

  dates.studentIDs().forEach((id) => {
    const user = state.entities.users[id]
    if (user) {
      users[id] = user.data
    }
  })

  return {
    assignment,
    users,
    courseColor,
  }
}
