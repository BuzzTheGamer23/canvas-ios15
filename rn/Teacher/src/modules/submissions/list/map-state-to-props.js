//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import type {
  SubmissionListDataProps,
} from './submission-prop-types'
import { getSubmissionsProps } from './get-submissions-props'
import shuffle from 'knuth-shuffle-seeded'
import {
  getGroupSubmissionProps,
} from '../../groups/submissions/get-group-submission-props'

type RoutingProps = {
  courseID: string,
  assignmentID: string,
}

export function mapStateToProps ({ entities }: AppState, { courseID, assignmentID }: RoutingProps): SubmissionListDataProps {
  // submissions
  const assignmentContent = entities.assignments[assignmentID]
  let pointsPossible
  let groupAssignment = null
  if (assignmentContent && assignmentContent.data) {
    const a = assignmentContent.data
    if (a.group_category_id) {
      groupAssignment = {
        groupCategoryID: a.group_category_id,
        gradeIndividually: a.grade_group_students_individually,
      }
    }
    pointsPossible = assignmentContent.data.points_possible
  }

  let submissions
  if (groupAssignment != null && !groupAssignment.gradeIndividually) {
    submissions = getGroupSubmissionProps(entities, courseID, assignmentID)
  } else {
    submissions = getSubmissionsProps(entities, courseID, assignmentID)
  }

  const courseContent = entities.courses[courseID]
  let courseColor = '#FFFFFF'
  if (courseContent && courseContent.color) {
    courseColor = courseContent.color
  }

  let courseName = ''
  if (courseContent && courseContent.course) {
    courseName = courseContent.course.name
  }

  let anonymous = !!assignmentContent && assignmentContent.anonymousGradingOn
  let muted = !!assignmentContent && assignmentContent.data.muted

  let assignmentName = ''
  if (assignmentContent && assignmentContent.data) {
    assignmentName = assignmentContent.data.name
  }
  let course = null
  if (courseContent && courseContent.course) {
    course = courseContent.course
  }

  return {
    groupAssignment,
    courseColor,
    courseName,
    pointsPossible,
    pending: submissions.pending,
    submissions: anonymous ? shuffle(submissions.submissions.slice(), assignmentID) : submissions.submissions,
    anonymous,
    muted,
    assignmentName,
    course,
  }
}
