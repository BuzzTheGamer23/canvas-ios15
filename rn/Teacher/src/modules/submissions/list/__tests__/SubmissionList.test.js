//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import React from 'react'
import { AlertIOS } from 'react-native'
import {
  SubmissionList,
  refreshSubmissionList,
  shouldRefresh,
} from '../SubmissionList'
import renderer from 'react-test-renderer'
import { submissionProps } from './map-state-to-props.test'
import setProps from '../../../../../test/helpers/setProps'
import cloneDeep from 'lodash/cloneDeep'

const template = {
  ...require('../../../../__templates__/helm'),
  ...require('../__templates__/submission-props'),
  ...require('../../../../__templates__/course'),
}

jest
  .mock('../../../../routing')
  .mock('AlertIOS', () => ({
    alert: jest.fn(),
  }))
  .mock('knuth-shuffle-seeded', () => jest.fn())

const props = {
  submissions: submissionProps,
  pending: false,
  courseID: '12',
  assignmentID: '32',
  courseColor: '#ddd',
  courseName: 'fancy name',
  refreshSubmissions: jest.fn(),
  refreshEnrollments: jest.fn(),
  refreshGroupsForCourse: jest.fn(),
  shouldRefresh: false,
  refreshing: false,
  refresh: jest.fn(),
  anonymous: false,
  muted: false,
  assignmentName: 'Blah',
  course: template.course(),
  groupAssignment: null,
}

beforeEach(() => jest.resetAllMocks())

test('SubmissionList loaded', () => {
  const tree = renderer.create(
    <SubmissionList {...props} navigator={template.navigator()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SubmissionList shows loading indicator', () => {
  const tree = renderer.create(
    <SubmissionList {...props} navigator={template.navigator()} pending={true} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SubmissionList loaded with nothing and it should not explode, and then props should be set and it should be great', () => {
  const tree = renderer.create(
    <SubmissionList {...props} navigator={template.navigator()} />
  )
  expect(tree).toBeDefined()
  setProps(tree, props)
  expect(tree.getInstance().state.submissions).toEqual(props.submissions)
})

test('SubmissionList select filter function', () => {
  const expandedProps = cloneDeep(props)
  expandedProps.submissions = expandedProps.submissions.concat([
    template.submissionProps({
      name: 'S5',
      status: 'submitted',
      grade: '60',
      score: 60,
      userID: '5',
    }),
    template.submissionProps({
      name: 'S6',
      status: 'submitted',
      grade: '30',
      score: 30,
      userID: '6',
    }),
  ])

  const instance = renderer.create(
    <SubmissionList { ...expandedProps } navigator={template.navigator()} />
  ).getInstance()

  instance.updateFilter({
    filter: instance.filterOptions[0],
  })
  expect(instance.state.submissions).toMatchObject(expandedProps.submissions)

  instance.updateFilter({
    filter: instance.filterOptions[1],
  })
  expect(instance.state.submissions).toMatchObject([
    {
      status: 'late',
    },
  ])

  instance.updateFilter({
    filter: instance.filterOptions[2],
  })
  expect(instance.state.submissions).toMatchObject([
    {
      grade: 'not_submitted',
    },
  ])

  instance.updateFilter({
    filter: instance.filterOptions[3],
  })
  expect(instance.state.submissions).toMatchObject([
    {
      grade: 'ungraded',
    },
  ])

  instance.updateFilter({
    filter: instance.filterOptions[5],
    metadata: 50,
  })
  expect(instance.state.submissions).toMatchObject([
    {
      score: 30,
      status: 'submitted',
    },
  ])

  // Scored more thanâ€¦
  instance.updateFilter({
    filter: instance.filterOptions[6],
    metadata: 50,
  })
  expect(instance.state.submissions).toMatchObject([
    {
      score: 60,
      status: 'submitted',
    },
  ])

  // Cancel
  instance.clearFilter()
  expect(instance.state.submissions).toMatchObject(expandedProps.submissions)
  instance.updateFilter({
    filter: instance.filterOptions[7],
  })
  expect(instance.state.submissions).toMatchObject(expandedProps.submissions)
})

test('SubmissionList renders correctly with a passed in filter', () => {
  const expandedProps = cloneDeep(props)
  expandedProps.submissions = expandedProps.submissions.concat([
    template.submissionProps({
      name: 'S5',
      status: 'submitted',
      grade: '60',
      score: 60,
      userID: '5',
    }),
  ])

  expandedProps.filterType = 'graded'

  const tree = renderer.create(
    <SubmissionList {...expandedProps} navigator={template.navigator()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('should refresh', () => {
  expect(shouldRefresh(props)).toEqual(true)
})

test('should navigate to submission settings', () => {
  let navigator = template.navigator({
    show: jest.fn(),
  })

  const tree = renderer.create(
    <SubmissionList {...props} navigator={navigator} />
  )
  tree.getInstance().openSettings()

  expect(navigator.show).toHaveBeenCalledWith('/courses/12/assignments/32/submission_settings', { modal: true })
})

test('should navigate to a submission', () => {
  let submission = props.submissions[0]
  let navigator = template.navigator({
    show: jest.fn(),
  })
  const tree = renderer.create(
    <SubmissionList {...props} navigator={navigator} />
  )
  tree.getInstance().navigateToSubmission(1)(submission.userID)

  expect(navigator.show).toHaveBeenCalledWith(
    '/courses/12/assignments/32/submissions/1',
    { modal: true, modalPresentationStyle: 'fullscreen' },
    { selectedFilter: undefined, studentIndex: 1 }
  )
})

test('shows message compose with correct data', () => {
  const expandedProps = cloneDeep(props)
  expandedProps.submissions = expandedProps.submissions.concat([
    template.submissionProps({
      name: 'S5',
      status: 'submitted',
      grade: '60',
      score: 60,
      userID: '5',
    }),
    template.submissionProps({
      name: 'S6',
      status: 'submitted',
      grade: '30',
      score: 30,
      userID: '6',
    }),
  ])
  let navigator = template.navigator({
    show: jest.fn(),
  })
  const tree = renderer.create(
    <SubmissionList {...expandedProps} navigator={navigator} />
  )
  let instance = tree.getInstance()

  let expectSubjectToBeCorrect = (subject: string) => {
    instance.messageStudentsWho()
    expect(navigator.show).toHaveBeenCalledWith('/conversations/compose', { modal: true }, {
      recipients: instance.state.submissions.map((submission) => {
        return { id: submission.userID, name: submission.name, avatar_url: submission.avatarURL }
      }),
      subject: subject,
      course: expandedProps.course,
      canAddRecipients: false,
      onlySendIndividualMessages: true,
    })
  }
  expect(navigator.show).toHaveBeenCalledW

  it('handles all submissions', () => {
    instance.updateFilter({
      filter: instance.filterOptions[0],
    })
    expectSubjectToBeCorrect('All submissions - Blah')
  })

  it('handles late submissions', () => {
    instance.updateFilter({
      filter: instance.filterOptions[1],
    })
    expectSubjectToBeCorrect('Submitted late - Blah')
  })

  it('handles un-submissions', () => {
    instance.updateFilter({
      filter: instance.filterOptions[2],
    })
    expectSubjectToBeCorrect("Haven't submitted yet - Blah")
  })

  it('handles un-gradedness', () => {
    instance.updateFilter({
      filter: instance.filterOptions[3],
    })
    expectSubjectToBeCorrect("Haven't been graded - Blah")
  })

  it('handles graded', () => {
    instance.updateFilter({
      filter: instance.filterOptions[4],
    })
    expectSubjectToBeCorrect('Graded - Blah')
  })

  it('handles scored less than', () => {
    instance.updateFilter({
      filter: instance.filterOptions[5],
      metadata: 50,
    })
    expectSubjectToBeCorrect('Scored less than 50 - Blah')
  })

  it('handles scored more than', () => {
    instance.updateFilter({
      filter: instance.filterOptions[6],
      metadata: 50,
    })
    expectSubjectToBeCorrect('Scored more than 50 - Blah')
  })
})

test('should prevent going to speed grader when offline', () => {
  let submission = props.submissions[0]
  let navigator = template.navigator({
    show: jest.fn(),
  })
  let tree = renderer.create(
    <SubmissionList {...props} navigator={navigator} />
  )
  tree.getInstance().setState({ isConnected: false })
  tree.getInstance().navigateToSubmission(1)(submission.userID)

  expect(AlertIOS.alert).toHaveBeenCalled()
  expect(navigator.show).not.toHaveBeenCalled()
})

test('refreshSubmissionList', () => {
  refreshSubmissionList(props)
  expect(props.refreshSubmissions).toHaveBeenCalledWith(props.courseID, props.assignmentID, false)
  expect(props.refreshEnrollments).toHaveBeenCalledWith(props.courseID)
})

test('refreshSubmissionList (group submissions)', () => {
  refreshSubmissionList({
    ...props,
    groupAssignment: { groupCategoryID: '2', gradeIndividually: false },
  })
  expect(props.refreshSubmissions).toHaveBeenCalledWith(props.courseID, props.assignmentID,
  true)
  expect(props.refreshGroupsForCourse).toHaveBeenCalledWith(props.courseID)
})
