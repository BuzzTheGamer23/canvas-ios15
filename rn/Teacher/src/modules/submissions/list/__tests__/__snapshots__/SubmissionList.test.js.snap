// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionList loaded 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "borderBottomColor": "lightgrey",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 16,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontFamily": ".SFUIDisplay-semibold",
            "fontSize": 20,
          },
          Object {
            "color": "#2d3b44",
            "fontSize": 20,
            "fontWeight": "600",
          },
        ]
      }
    >
      All Submissions
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="submission-list.filter"
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#ff0ee8",
                "fontSize": 14,
              },
              Object {
                "fontFamily": ".SFUIDisplay-bold",
              },
              Object {
                "color": "#ff0ee8",
              },
              Object {
                "marginBottom": 1,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
      >
        Filter
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "avatarURL": "http://www.fillmurray.com/100/100",
          "grade": "not_submitted",
          "name": "S1",
          "status": "none",
          "userID": "1",
        },
        Object {
          "avatarURL": "http://www.fillmurray.com/100/100",
          "grade": "excused",
          "name": "S2",
          "status": "none",
          "userID": "2",
        },
        Object {
          "avatarURL": "http://www.fillmurray.com/100/100",
          "grade": "ungraded",
          "name": "S3",
          "status": "late",
          "userID": "3",
        },
        Object {
          "avatarURL": "http://www.fillmurray.com/100/100",
          "grade": "A-",
          "name": "S4",
          "status": "submitted",
          "userID": "4",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onViewableItemsChanged={undefined}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={true}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    shouldItemUpdate={[Function]}
    testID="submission-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "borderBottomColor": "lightgrey",
              "borderBottomWidth": 0.5,
              "flex": 1,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            testID="submission-1"
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "http://www.fillmurray.com/100/100",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "borderRadius": 20,
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingLeft": 8,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#2D3B45",
                        "flex": 1,
                        "fontSize": 16,
                        "fontWeight": "600",
                      },
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  S1
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "fontSize": 14,
                          "paddingTop": 2,
                        },
                        Object {
                          "color": "#8B969E",
                        },
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                    ]
                  }
                >
                  No submission
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#c7c7cc",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                    "height": 10,
                    "marginLeft": 7,
                    "transform": Array [
                      Object {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 10,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "borderBottomColor": "lightgrey",
              "borderBottomWidth": 0.5,
              "flex": 1,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            testID="submission-2"
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "http://www.fillmurray.com/100/100",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "borderRadius": 20,
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingLeft": 8,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#2D3B45",
                        "flex": 1,
                        "fontSize": 16,
                        "fontWeight": "600",
                      },
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  S2
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "fontSize": 14,
                          "paddingTop": 2,
                        },
                        Object {
                          "color": "#8B969E",
                        },
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                    ]
                  }
                >
                  No submission
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#368BD8",
                        "fontSize": 14,
                      },
                      Object {
                        "alignSelf": "center",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                  ]
                }
              >
                Excused
              </Text>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#c7c7cc",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                    "height": 10,
                    "marginLeft": 7,
                    "transform": Array [
                      Object {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 10,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "borderBottomColor": "lightgrey",
              "borderBottomWidth": 0.5,
              "flex": 1,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            testID="submission-3"
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "http://www.fillmurray.com/100/100",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "borderRadius": 20,
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingLeft": 8,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#2D3B45",
                        "flex": 1,
                        "fontSize": 16,
                        "fontWeight": "600",
                      },
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  S3
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "fontSize": 14,
                          "paddingTop": 2,
                        },
                        Object {
                          "color": "#FC5E13",
                        },
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                    ]
                  }
                >
                  Late
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                color="#368BD8"
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "backgroundColor": "white",
                        "borderRadius": 9,
                        "borderWidth": 1,
                        "fontSize": 11,
                        "fontWeight": "500",
                        "overflow": "hidden",
                        "paddingBottom": 0,
                        "paddingLeft": 6,
                        "paddingRight": 6,
                        "paddingTop": 3,
                      },
                      Object {
                        "borderColor": "#368BD8",
                        "color": "#368BD8",
                      },
                      Object {
                        "alignSelf": "center",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay-medium",
                    },
                  ]
                }
              >
                UNGRADED
              </Text>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#c7c7cc",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                    "height": 10,
                    "marginLeft": 7,
                    "transform": Array [
                      Object {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 10,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "borderBottomColor": "lightgrey",
              "borderBottomWidth": 0.5,
              "flex": 1,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            testID="submission-4"
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "http://www.fillmurray.com/100/100",
                  }
                }
                style={
                  Object {
                    "alignSelf": "center",
                    "borderRadius": 20,
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "paddingLeft": 8,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#2D3B45",
                        "flex": 1,
                        "fontSize": 16,
                        "fontWeight": "600",
                      },
                      Object {
                        "fontFamily": ".SFUIDisplay-semibold",
                      },
                    ]
                  }
                >
                  S4
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "fontSize": 14,
                          "paddingTop": 2,
                        },
                        Object {
                          "color": "#07AF1F",
                        },
                      ],
                      Object {
                        "fontFamily": ".SFUIDisplay",
                      },
                    ]
                  }
                >
                  Submitted
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#368BD8",
                        "fontSize": 14,
                      },
                      Object {
                        "alignSelf": "center",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                  ]
                }
              >
                A-
              </Text>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#c7c7cc",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                    "height": 10,
                    "marginLeft": 7,
                    "transform": Array [
                      Object {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 10,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
